base/applications/rapps/appview.cpp

```diff
@@ -1382,7 +1382,21 @@ BOOL CAppsListView::AddInstalledApplication(CInstalledApplicationInfo *InstAppIn
         return FALSE;
     }
 
-    HICON hIcon = (HICON)LoadIconW(hInst, MAKEINTRESOURCEW(IDI_MAIN));
+    /* Load icon from registry */
+    HICON hIcon = NULL;
+    ATL::CStringW szIconPath;
+    if (InstAppInfo->RetrieveIcon(szIconPath))
+    {
+        hIcon = (HICON)ExtractIconW(hInst,
```

Comment:
You might want to play with `PathParseIconLocationW` ;)

Reply:
or just extract it manually from the CStringW

Reply:
If the string also starts with an `@` you might consider passing it to `SHLoadIndirectString`

Reply:
Fixed, thanks @learn-more! :+1: 
`PathParseIconLocationW` works as well for me with zero-terminated strings.
But I don't remember any software, which has `@` at the start of the string from `DisplayIcon` value. Is there any? :thinking: